### ISSUE TITLE
feat: add Table.cast(ibis.Schema | any_input_that_ibis.schema()_accepts)

### ISSUE THREAD

**Issue by NickCrews** [ibis-project/ibis#5610]
### Is your feature request related to a problem?

It's currently some boilerplate to cast an entire table to a schema. I have to go through column by column.

### Describe the solution you'd like

I suggested .cast() because that mirrors the .cast() that is available for columns and scalars.

You should be able to pass a dict, an iterable to pairs, or anything that ibis.schema() accepts as an arg:

```python
def cast(self, schema_ish):
    if isinstance(schema_ish, Schema):
        schema = schema_ish
    else:
        schema = ibis.schema(schema_ish)
    # continues...
```

### What version of ibis are you running?

master

### What backend(s) are you using, if any?

all

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by cpcloud** [comment]
Seems like a good idea! Thanks for the issue.

**Comment by NickCrews** [comment]
hmmm, I'm thinking about what should happen with Tables that have too-few or too-many columns. I propose 
```python
def cast(
    self,
    schema_ish,
    on_extra: Literal["error"] | Literal["keep"] | Literal["drop"] = "error",
    on_missing: Literal["error"] | Literal["add"] = "error",
) -> Table
```
# on_extra
If there are columns in table that aren't in schema
- keep -> keep the existing columns unchanged
- drop -> drop them
- error -> raise

# on_missing
if there are columns in schema that aren't in table:
- error -> raise
- add -> add the missing columns, filled with null

**Comment by cpcloud** [comment]
How about we just keep it simple and provide no additional arguments, with the following invariant:

`table.cast` always returns a table with the same column names as the caller

This would mean:

1. Non-existent columns are ignored.
2. Schemas that don't include all columns operate on a subset.

Mixing other projection operations like keeping, dropping or literal filling are both out of scope. A more flexible drop and literal filling API will be available after #5598 lands.
